### PR TITLE
Better interaction target

### PR DESCRIPTION
**Pull request by Bod9001**
because the client is sending relative position but it also has client predictions are that means, 
when the relative position is applied on the Server State is incorrect, so this sends the local position where the player is Clicking, 
providing better aiming

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Input System/CablePlacementVisualisation.cs [hunk 1]
```diff
@@ -151,8 +151,9 @@ private void UpdateMe()
 	private void Build()
 	{
 		if (startPoint == endPoint || target == null) return;
+		var Position = cablePlacementVisualisation.transform.position + new Vector3(0.5f, 0.5f, 0);
 
-		Vector2 targetVector = (connectionPointRenderers[endPoint].transform.position - transform.position);
+		Vector2 targetVector = Position.ToLocal(PlayerManager.LocalPlayer.RegisterTile().Matrix);; // transform.position ( - transform.position); //TODO? what? is this
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/CablePlacementVisualisation.cs]:
```suggestion
		Vector2 targetVector = Position.ToLocal(PlayerManager.LocalPlayer.RegisterTile().Matrix); // transform.position ( - transform.position); //TODO? what? is this
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/AimApply.cs [hunk 2]
```diff
@@ -31,17 +39,14 @@ public class AimApply : BodyPartTargetedInteraction
 	/// <summary>
 	/// Targeted world position deduced from target vector and performer position.
 	/// </summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/AimApply.cs]:
```suggestion
	/// <summary>Target world position calculated from matrix local position.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/AimApply.cs [hunk 3]
```diff
@@ -31,17 +39,14 @@ public class AimApply : BodyPartTargetedInteraction
...
-	public Vector2 WorldPositionTarget => (Vector2)Performer.transform.position + targetVector;
+	public Vector2 WorldPositionTarget => (Vector2)targetPosition.To3().ToWorld(Performer.RegisterTile().Matrix);
 
-	/// <summary>
-	/// Vector pointing from the performer to the targeted position. Set to Vector2.zero if aiming at self.
-	/// </summary>
-	public Vector2 TargetVector => targetVector;
+	public Vector2 TargetVector => targetPosition - Performer.transform.localPosition.To2() ;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/AimApply.cs]:
```suggestion
	/// <summary>Vector pointing from the performer's position to the target position.</summary>
	public Vector2 TargetVector => targetPosition - Performer.transform.localPosition.To2();
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/ConnectionApply.cs [hunk 4]
```diff
@@ -19,17 +19,17 @@ public class ConnectionApply : TargetedInteraction
...
 
 	/// <summary>
 	/// Targeted world position deduced from target vector and performer position.
 	/// </summary>
-	public Vector2 WorldPositionTarget => (Vector2)Performer.transform.position + targetVector;
+	public Vector2 WorldPositionTarget => (Vector2)targetPosition.To3().ToWorld(Performer.RegisterTile().Matrix);
 
 	/// <summary>
 	/// Vector pointing from the performer to the targeted position. Set to Vector2.zero if aiming at self.
 	/// </summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/ConnectionApply.cs]:
```suggestion
	/// <summary>Requested local position target.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/AimApply.cs [hunk 5]
```diff
@@ -15,9 +15,17 @@ public class AimApply : BodyPartTargetedInteraction
...
 	private readonly MouseButtonState mouseButtonState;
-	private readonly Vector2 targetVector;
+	private readonly Vector2 targetPosition;
+	public readonly Vector2 originatorPosition; //Warning this includes client prediction unsafe
 	private readonly ItemSlot handSlot;
 
+
+	/// <summary>
+	/// Vector pointing from the performer to the targeted position. Set to Vector2.zero if aiming at self.
+	/// </summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/AimApply.cs]:
```suggestion
/// <summary>Requested local position target.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/ConnectionApply.cs [hunk 6]
```diff
@@ -19,17 +19,17 @@ public class ConnectionApply : TargetedInteraction
 
 	private readonly Connection connectionPointA, connectionPointB;
 
-	private readonly Vector2 targetVector;
+	private readonly Vector2 targetPosition;
 
 	/// <summary>
 	/// Targeted world position deduced from target vector and performer position.
 	/// </summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/ConnectionApply.cs]:
```suggestion
	/// <summary>Target world position calculated from matrix local position.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/PositionalHandApply.cs [hunk 7]
```diff
@@ -11,29 +11,32 @@ public class PositionalHandApply : HandApply
 	private static readonly PositionalHandApply Invalid
 			= new PositionalHandApply(null, null, null, Vector2.zero, null, Intent.Help, BodyPartType.None, false);
 
+	/// <summary>Target position, the Local position that the performer is pointing at</summary>
+	public Vector2 TargetPosition { get; protected set; }
+
 	/// <summary>Vector pointing from the performer to the targeted position. Set to Vector2.zero if aiming at self.</summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/PositionalHandApply.cs]:
```suggestion
	/// <summary>Vector pointing from the performer's position to the target position.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/PositionalHandApply.cs [hunk 8]
```diff
@@ -11,29 +11,32 @@ public class PositionalHandApply : HandApply
...
-	public Vector2 TargetVector { get; protected set; }
+	public Vector2 TargetVector => TargetPosition - Performer.transform.localPosition.To2();
 
 	/// <summary>Targeted world position deduced from target vector and performer position.</summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/PositionalHandApply.cs]:
```suggestion
	/// <summary>Target world position calculated from matrix local position.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileApply.cs [hunk 9]
```diff
@@ -38,17 +38,20 @@ public class TileApply : Interaction
 	/// </summary>
 	public GameObject HandObject => UsedObject;
 
-	private readonly Vector2 targetVector;
+	private readonly Vector2 targetPosition;
+
 
 	/// <summary>
 	/// Targeted world position deduced from target vector and performer position.
 	/// </summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileApply.cs]:
```suggestion
	/// <summary>Target world position calculated from matrix local position.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileApply.cs [hunk 10]
```diff
@@ -38,17 +38,20 @@ public class TileApply : Interaction
...
-	public Vector2 WorldPositionTarget => (Vector2)Performer.transform.position + targetVector;
+	public Vector2 WorldPositionTarget => TargetPosition.To3().ToWorld(Performer.RegisterTile().Matrix);
 
 	/// <summary>
 	/// Vector pointing from the performer to the targeted position. Set to Vector2.zero if aiming at self.
 	/// </summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileApply.cs]:
```suggestion
	/// <summary>Requested local position target.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileApply.cs [hunk 11]
```diff
@@ -38,17 +38,20 @@ public class TileApply : Interaction
...
-	public Vector2 TargetVector => targetVector;
+	public Vector2 TargetPosition => targetPosition;
+
+	public Vector2 TargetVector => Performer.transform.localPosition.To2() - targetPosition;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileApply.cs]:
```suggestion
	/// <summary>Vector pointing from the performer's position to the target position.</summary>
	public Vector2 TargetVector => targetPosition - Performer.transform.localPosition.To2();
```

This one was backwards, right?

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileMouseDrop.cs [hunk 12]
```diff
@@ -24,24 +24,24 @@ public class TileMouseDrop : Interaction
 	/// </summary>
 	public BasicTile BasicTile => basicTile;
 
-	private readonly Vector2 targetVector;
+	private readonly Vector2 targetPosition;
 
 	/// <summary>
 	/// Targeted world position deduced from target vector and performer position.
 	/// </summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileMouseDrop.cs]:
```suggestion
	/// <summary>Target world position calculated from matrix local position.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileMouseDrop.cs [hunk 13]
```diff
@@ -24,24 +24,24 @@ public class TileMouseDrop : Interaction
...
-	public Vector2 WorldPositionTarget => (Vector2)Performer.transform.position + targetVector;
+	public Vector2 WorldPositionTarget => (Vector2)targetPosition.To3().ToWorld(Performer.RegisterTile().Matrix);
 
 	/// <summary>
 	/// Vector pointing from the performer to the targeted position. Set to Vector2.zero if aiming at self.
 	/// </summary>
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileMouseDrop.cs]:
```suggestion
	/// <summary>Requested local position target.</summary>
```

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/AimApply.cs [hunk 14]
```diff
@@ -76,10 +82,13 @@ public static AimApply ByLocalPlayer(MouseButtonState buttonState)
 			PLAYER_LAYER_MASK = LayerMask.GetMask("Players");
 		}
 
-		var targetVector = (Vector2) MouseUtils.MouseToWorldPos() -
-		                   (Vector2) PlayerManager.LocalPlayer.transform.position;
+		var targetPosition = MouseUtils.MouseToWorldPos().ToLocal(PlayerManager.LocalPlayer.RegisterTile().Matrix).To2();
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/AimApply.cs]: What Codacy pointed out probably makes a compiler warning.

**Review by NoooneyDude — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/ConnectionApply.cs [hunk 15]
```diff
@@ -69,11 +65,20 @@ public class ConnectionApply : TargetedInteraction
 	/// <param name="wireEndB">cable end connection</param>
 	/// <param name="worldPositionTarget">position of target tile (world space)</param>
 	/// <returns></returns>
-	public static ConnectionApply ByLocalPlayer(GameObject targetObject, Connection wireEndA, Connection wireEndB, Vector2? targetVector)
+	public static ConnectionApply ByLocalPlayer(GameObject targetObject, Connection wireEndA, Connection wireEndB, Vector3? IntargetVector)
 	{
 		if (PlayerManager.LocalPlayerScript.IsGhost) return Invalid;
+		Vector3 targetVec = Vector3.zero;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Interactions/ConnectionApply.cs]:
```suggestion
		Vector3 targetVec;
```

**Review by NoooneyDude — APPROVED** [review]

File: UnityProject/Assets/Scripts/Core/Input System/CablePlacementVisualisation.cs [hunk 16]
```diff
@@ -151,8 +151,9 @@ private void UpdateMe()
 	private void Build()
 	{
 		if (startPoint == endPoint || target == null) return;
+		var Position = cablePlacementVisualisation.transform.position + new Vector3(0.5f, 0.5f, 0);
 
-		Vector2 targetVector = (connectionPointRenderers[endPoint].transform.position - transform.position);
+		Vector2 targetVector = Position.ToLocal(PlayerManager.LocalPlayer.RegisterTile().Matrix); // transform.position ( - transform.position); //TODO? what? is this
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Core/Input System/CablePlacementVisualisation.cs:156]: Have you tested placing cables after this change?

Don't really get the TODO, it's just getting the direction vector between the points?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Input System/CablePlacementVisualisation.cs:156]: it works fine

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**